docs/content/development/extensions-core/sample-client-sslcontext.md

```diff
@@ -0,0 +1,27 @@
+---
+layout: doc_page
+---
+
+## Sample SSLContext Provider Module
+
+This module contains a sample implementation of [SSLContext](http://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLContext.html)
+that will be injected to be used with HttpClient that Druid nodes use internally to communicate with each other. To learn more about
+Java's SSL support, please refer to [this](http://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/JSSERefGuide.html) guide.
+
+# Configuration
+
+|Property|Description|Default|Required|
+|--------|-----------|-------|--------|
+|`druid.client.https.algorithm`|SSL Algorithm/Protocol to use.|`null`|yes|
+|`druid.client.https.trustStoreType`|The type of the key store where trusted root certificates are stored.|`java.security.KeyStore.getDefaultType()`|yes|
+|`druid.client.https.trustStorePath`|The file path or URL of the TLS/SSL Key store where trusted root certificates are stored.|`null`|yes|
+|`druid.client.https.trustStoreAlgorithm`|Algorithm to be used by TrustManager to validate certificate chains|`javax.net.ssl.TrustManagerFactory.getDefaultAlgorithm()`|yes|
+
+All these configs must be set to some valid value. Apart from these properties, `trustStorePassword` is required to access trustStore.
+This password will be fetched using a `PasswordProvider` implementation.
+For example, `DefaultPasswordProvider` will look for `druid.password.trustStorePassword` runtime property to get these passwords.
+`EnvironmentVariablePasswordProvider` will directly look for value of `trustStorePassword` environment variable.
```

Comment:
`trustStorePassword` is the key name used to call `getPassword(String key)` method on `PasswordProvider` interface. So the key name sounds ok for `DefaultPasswordProvider` as it will actually look for `druid.password.trustStorePassword`. For `EnvironmentVariablePasswordProvider` we can have a convention which applies toUpper function to the key name and prefixes it with `DRUID_` to get the env name to look for, so it will look for "DRUID_TRUSTSTOREPASSWORD". What do you think ?